presto-spi/src/main/java/com/facebook/presto/spi/plan/IntersectNode.java

```diff
@@ -11,34 +11,32 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.facebook.presto.sql.planner.plan;
+package com.facebook.presto.spi.plan;
 
-import com.facebook.presto.spi.plan.PlanNode;
-import com.facebook.presto.spi.plan.PlanNodeId;
 import com.facebook.presto.spi.relation.VariableReferenceExpression;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.google.common.collect.ListMultimap;
 
 import javax.annotation.concurrent.Immutable;
 
+import java.util.LinkedHashMap;
 import java.util.List;
 
 @Immutable
-public class IntersectNode
+public final class IntersectNode
         extends SetOperationNode
 {
     @JsonCreator
     public IntersectNode(
             @JsonProperty("id") PlanNodeId id,
             @JsonProperty("sources") List<PlanNode> sources,
-            @JsonProperty("outputToInputs") ListMultimap<VariableReferenceExpression, VariableReferenceExpression> outputToInputs)
+            @JsonProperty("outputToInputs") LinkedHashMap<VariableReferenceExpression, List<VariableReferenceExpression>> outputToInputs)
```

Comment:
Use `Map<VariableReferenceExpression, List<VariableReferenceExpression>>` ?

Reply:
Use `LinkedHashMap` to enforce callsite to be aware of the order.

Reply:
Yes, but do we really need the order? If yes, I think @rongrong's suggestion might be better.